unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PlayerInfo/PassportPlayerInfoComponentView.cs

```diff
@@ -128,6 +128,37 @@ public void InitializeJumpInButton(IFriendsController friendsController, string
             }
         }
 
+        public void SetFriendStatus(FriendshipStatus friendStatus)
+        {
+            areFriends = friendStatus == FriendshipStatus.FRIEND;
+
+            if(isBlocked)
+                return;
+
+            switch (friendStatus)
+            {
+                case FriendshipStatus.NOT_FRIEND:
+                    DisableAllFriendFlowButtons();
+                    addFriendButton.gameObject.SetActive(true);
+                    break;
+                case FriendshipStatus.FRIEND:
+                    DisableAllFriendFlowButtons();
+                    alreadyFriendsButton.gameObject.SetActive(true);
+                    break;
+                case FriendshipStatus.REQUESTED_FROM:
+                    DisableAllFriendFlowButtons();
+                    acceptFriendButton.gameObject.SetActive(true);
+                    break;
+                case FriendshipStatus.REQUESTED_TO:
+                    DisableAllFriendFlowButtons();
+                    cancelFriendRequestButton.gameObject.SetActive(true);
+                    break;
+                default:
+                    break;
+            }
```

Comment:
Creating a dictionary would remove this big switch (and duplication inside it)

Reply:
done